ftw/subsite/viewlets/subsitelogoviewlet.py

```diff
@@ -18,29 +20,35 @@ def __init__(self, context, request, view, manager=None):
 
     def update(self):
         super(SubsiteLogoViewlet, self).update()
-
-        portal = self.portal_state.portal()
-        self.navigation_root_url = self.portal_state.navigation_root_url()
-
-        subsite_logo = getattr(self.context, 'getLogo', None)
-        parent = self.context.aq_inner.aq_parent
-        in_factory = IFactoryTempFolder.providedBy(parent)
-
-        if subsite_logo and subsite_logo() and not in_factory:
-            # we are in a subsite
-            context = self.context
-            if not IContentish.providedBy(context):
-                context = context.aq_parent
-            navigation_root_path = getNavigationRoot(context)
-            scale = portal.restrictedTraverse(
-                navigation_root_path + '/@@images')
+        portal = api.portal.get()
+        subsite = self.get_nearest_subsite_with_logo()
+        if subsite:
+            subsite_image_scales_path = '/'.join(
+                list(subsite.getPhysicalPath()) + ['@@images']
+            )
+            scales = portal.restrictedTraverse(subsite_image_scales_path)
             # XXX Use own scale
-            self.logo_tag = scale.scale('logo', scale="logo").tag()
-            self.title = self.context.restrictedTraverse(
-                getNavigationRoot(self.context)).Title()
+            self.logo_tag = scales.scale('logo', scale="logo").tag()
+            self.title = subsite.Title()
             self.is_subsitelogo = True
         else:
             # standard plone logo
             logoName = portal.restrictedTraverse('base_properties').logoName
             self.logo_tag = portal.restrictedTraverse(logoName).tag()
             self.title = self.portal_state.portal_title()
+
+    def get_nearest_subsite_with_logo(self):
+        obj = self.context
+        while not IPloneSiteRoot.providedBy(obj):
```

Comment:
-> `while obj and not IPloneSiteRoot.providedBy(obj):`
Also check for obj beeing truthy: 
If the viewlet is (accidentally) rendered on a context not within a Plone Site, it will never exit the `while` loop.
Although this is not an expected situation, it occurs from time to time, e.g. when running a maintenance script with `bin/instance run`  or so.

When the top of the acquisition chain is reached, the next parent will be None. So the parent of the `app` is `None`, which will break the loop.